<a href="https://colab.research.google.com/github/Said-Akbar/Data-science/blob/master/Ebay_used_cars.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Ebay dataset of used cars

In this project, we are going to analyze a dataset of used cars from Ebay, which was scraped and cleaned to use by a Kaggle user. The link for the dataset is [here](https://www.kaggle.com/orgesleka/used-cars-database/data). The dataset contains around 50,000 records and 20 columns that includes car price, condition, year and more. Because the dataset is not clean, we will first do data cleaning and then dive into analyses. 

In [0]:
import numpy as np
import pandas as pd

autos = pd.read_csv("autos.csv", encoding="Latin-1")

In [0]:
autos

dateCrawled                                               name  \
0      2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1      2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2      2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3      2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4      2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   
5      2016-03-21 13:47:45  Chrysler_Grand_Voyager_2.8_CRD_Aut.Limited_Sto...   
6      2016-03-20 17:55:21  VW_Golf_III_GT_Special_Electronic_Green_Metall...   
7      2016-03-16 18:55:19                               Golf_IV_1.9_TDI_90PS   
8      2016-03-22 16:51:34                                         Seat_Arosa   
9      2016-03-16 13:47:02          Renault_Megane_Scenic_1.6e_RT_Klimaanlage   
10     2016-03-15 01:41:36                       VW_Golf_Tuning_in_siber/grau   
11     2016-03-16 18:45:34                         Mercedes_A140_Motorschaden   
12     2016-03-31 19:48:22  Smart_smart_fortwo_coupe_softouch_pure_MHD_Pan...   
13     2016-03-23 10:48:32                                 Audi_A3_1.6_tuning   
14     2016-03-23 11:50:46  Renault_Clio_3__Dynamique_1.2__16_V;_viele_Ver...   
15     2016-04-01 12:06:20        Corvette_C3_Coupe_T_Top_Crossfire_Injection   
16     2016-03-16 14:59:02                                Opel_Vectra_B_Kombi   
17     2016-03-29 11:46:22                          Volkswagen_Scirocco_2_G60   
18     2016-03-26 19:57:44               Verkaufen_mein_bmw_e36_320_i_touring   
19     2016-03-17 13:36:21        mazda_tribute_2.0_mit_gas_und_tuev_neu_2018   
20     2016-03-05 19:57:31  Audi_A4_Avant_1.9_TDI_*6_Gang*AHK*Klimatronik*...   
21     2016-03-06 19:07:10                      Porsche_911_Carrera_4S_Cabrio   
22     2016-03-28 20:50:54                               MINI_Cooper_S_Cabrio   
23     2016-03-10 19:55:34   Peugeot_Boxer_2_2_HDi_120_Ps_9_Sitzer_inkl_Klima   
24     2016-04-03 11:57:02                         BMW_535i_xDrive_Sport_Aut.   
25     2016-03-21 21:56:18  Ford_escort_kombi_an_bastler_mit_ghia_ausstattung   
26     2016-04-03 22:46:28                                Volkswagen_Polo_Fox   
27     2016-03-27 18:45:01             Hat_einer_Ahnung_mit_Ford_Galaxy_HILFE   
28     2016-03-19 21:56:19                                      MINI_Cooper_D   
29     2016-04-02 12:45:44  Mercedes_Benz_E_320_T_CDI_Avantgarde_DPF7_Sitz...   
...                    ...                                                ...   
49970  2016-03-21 22:47:37  c4_Grand_Picasso_mit_Automatik_Leder_Navi_Temp...   
49971  2016-03-29 14:54:12                                         W.Lupo_1.0   
49972  2016-03-26 22:25:23          Mercedes_Benz_Vito_115_CDI_Extralang_Aut.   
49973  2016-03-27 05:32:39                   Mercedes_Benz_SLK_200_Kompressor   
49974  2016-03-20 10:52:31  Golf_1_Cabrio_Tuev_Neu_viele_Extras_alles_eing...   
49975  2016-03-27 20:51:39          Honda_Jazz_1.3_DSi_i_VTEC_IMA_CVT_Comfort   
49976  2016-03-19 18:56:05  Audi_80_Avant_2.6_E__Vollausstattung!!_Einziga...   
49977  2016-03-31 18:37:18                        Mercedes_Benz_C200_Cdi_W203   
49978  2016-04-04 10:37:14                        Mercedes_Benz_E_200_Classic   
49979  2016-03-20 18:38:40                      Volkswagen_Polo_1.6_TDI_Style   
49980  2016-03-12 10:55:54                            Ford_Escort_Turnier_16V   
49981  2016-03-15 09:38:21             Opel_Astra_Kombi_mit_Anhaengerkupplung   
49982  2016-03-29 18:51:08            Skoda_Fabia_4_Tuerer_Bj:2004__85.000Tkm   
49983  2016-03-06 12:43:04                                      Ford_focus_99   
49984  2016-03-31 22:48:48  Student_sucht_ein__Anfaengerauto___ab_2000_BJ_...   
49985  2016-04-02 16:38:23                          Verkaufe_meinen_vw_vento!   
49986  2016-04-04 20:46:02  Chrysler_300C_3.0_CRD_DPF_Automatik_Voll_Ausst...   
49987  2016-03-22 20:47:27  Audi_A3_Limousine_

In [0]:
autos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50000 entries, 0 to 49999
Data columns (total 20 columns):
dateCrawled            50000 non-null object
name                   50000 non-null object
seller                 50000 non-null object
offerType              50000 non-null object
price                  50000 non-null object
abtest                 50000 non-null object
vehicleType            44905 non-null object
yearOfRegistration     50000 non-null int64
gearbox                47320 non-null object
powerPS                50000 non-null int64
model                  47242 non-null object
odometer               50000 non-null object
monthOfRegistration    50000 non-null int64
fuelType               45518 non-null object
brand                  50000 non-null object
notRepairedDamage      40171 non-null object
dateCreated            50000 non-null object
nrOfPictures           50000 non-null int64
postalCode             50000 non-null int64
lastSeen               50000 non-null obj

In [0]:
autos.head()

dateCrawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offerType   price   abtest vehicleType  yearOfRegistration  \
0  privat   Angebot  $5,000  control         bus                2004   
1  privat   Angebot  $8,500  control   limousine                1997   
2  privat   Angebot  $8,990     test   limousine                2009   
3  privat   Angebot  $4,350  control  kleinwagen                2007   
4  privat   Angebot  $1,350     test       kombi                2003   

     gearbox  powerPS   model   odometer  monthOfRegistration fuelType  \
0    manuell      158  andere  150,000km                    3      lpg   
1  automatik      286     7er  150,000km                    6   benzin   
2    manuell      102    golf   70,000km                    7   benzin   
3  automatik       71  fortwo   70,000km                    6   benzin   
4    manuell        0   focus  150,000km                    7   benzin   

        brand notRepairedDamage          dateCreated  nrOfPictures  \
0     peugeot              nein  2016-03-26 00:00:00             0   
1         bmw              nein  2016-04-04 00:00:00             0   
2  volkswagen              nein  2016-03-26 00:00:00             0   
3       smart              nein  2016-03-12 00:00:00             0   
4        ford              nein  2016-04-01 00:00:00             0   

   postalCode             lastSeen  
0       79588  2016-04-06 06:45:54  
1       71034  2016-04-06 14:45:08  
2       35394  2016-04-06 20:15:37  
3       33729  2016-03-15 03:16:28  
4       39218  2016-04-01 14:38:50

Above summary tables show that the dataset was crawled in 2016 and cars were listed online also in 2016. There are some missing values in'vehicleType', 'Gearbox', 'model', 'fuelType' and 'notRepairedDamage'. Although prices are shown in US dollars, other columns information is presented in German, not English. Moreover, only 5 columns are of numeric type and the rest of the 15 columns are string type.

Column names are in CamelCase. But we need to convert them into snake_case.

In [0]:
autos.columns

Index(['dateCrawled', 'name', 'seller', 'offerType', 'price', 'abtest',
       'vehicleType', 'yearOfRegistration', 'gearbox', 'powerPS', 'model',
       'odometer', 'monthOfRegistration', 'fuelType', 'brand',
       'notRepairedDamage', 'dateCreated', 'nrOfPictures', 'postalCode',
       'lastSeen'],
      dtype='object')

In [0]:
new_columns = ['date_crawled', 'name', 'seller', 'offer_type', 'price', 'abtest',
       'vehicle_type', 'registration_year', 'gearbox', 'power_ps', 'model',
       'odometer', 'registration_month', 'fuel_type', 'brand',
       'unrepaired_damage', 'ad_created', 'n_pictures', 'postal_code',
       'last_seen']

In [0]:
autos.columns = new_columns

In [0]:
autos.head()

date_crawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offer_type   price   abtest vehicle_type  registration_year  \
0  privat    Angebot  $5,000  control          bus               2004   
1  privat    Angebot  $8,500  control    limousine               1997   
2  privat    Angebot  $8,990     test    limousine               2009   
3  privat    Angebot  $4,350  control   kleinwagen               2007   
4  privat    Angebot  $1,350     test        kombi               2003   

     gearbox  power_ps   model   odometer  registration_month fuel_type  \
0    manuell       158  andere  150,000km                   3       lpg   
1  automatik       286     7er  150,000km                   6    benzin   
2    manuell       102    golf   70,000km                   7    benzin   
3  automatik        71  fortwo   70,000km                   6    benzin   
4    manuell         0   focus  150,000km                   7    benzin   

        brand unrepaired_damage           ad_created  n_pictures  postal_code  \
0     peugeot              nein  2016-03-26 00:00:00           0        79588   
1         bmw              nein  2016-04-04 00:00:00           0        71034   
2  volkswagen              nein  2016-03-26 00:00:00           0        35394   
3       smart              nein  2016-03-12 00:00:00           0        33729   
4        ford              nein  2016-04-01 00:00:00           0        39218   

             last_seen  
0  2016-04-06 06:45:54  
1  2016-04-06 14:45:08  
2  2016-04-06 20:15:37  
3  2016-03-15 03:16:28  
4  2016-04-01 14:38:50

We changed column names to snake_case and shortened some of the names so that column names are easier to access.

In [0]:
autos.describe(include='all')

date_crawled         name  seller offer_type  price abtest  \
count                 50000        50000   50000      50000  50000  50000   
unique                48213        38754       2          2   2357      2   
top     2016-03-11 22:38:16  Ford_Fiesta  privat    Angebot     $0   test   
freq                      3           78   49999      49999   1421  25756   
mean                    NaN          NaN     NaN        NaN    NaN    NaN   
std                     NaN          NaN     NaN        NaN    NaN    NaN   
min                     NaN          NaN     NaN        NaN    NaN    NaN   
25%                     NaN          NaN     NaN        NaN    NaN    NaN   
50%                     NaN          NaN     NaN        NaN    NaN    NaN   
75%                     NaN          NaN     NaN        NaN    NaN    NaN   
max                     NaN          NaN     NaN        NaN    NaN    NaN   

       vehicle_type  registration_year  gearbox      power_ps  model  \
count         44905       50000.000000    47320  50000.000000  47242   
unique            8                NaN        2           NaN    245   
top       limousine                NaN  manuell           NaN   golf   
freq          12859                NaN    36993           NaN   4024   
mean            NaN        2005.073280      NaN    116.355920    NaN   
std             NaN         105.712813      NaN    209.216627    NaN   
min             NaN        1000.000000      NaN      0.000000    NaN   
25%             NaN        1999.000000      NaN     70.000000    NaN   
50%             NaN        2003.000000      NaN    105.000000    NaN   
75%             NaN        2008.000000      NaN    150.000000    NaN   
max             NaN        9999.000000      NaN  17700.000000    NaN   

         odometer  registration_month fuel_type       brand unrepaired_damage  \
count       50000        50000.000000     45518       50000             40171   
unique         13                 NaN         7          40                 2   
top     150,000km                 NaN    benzin  volkswagen              nein   
freq        32424                 NaN     30107       10687             35232   
mean          NaN            5.723360       NaN         NaN               NaN   
std           NaN            3.711984       NaN         NaN               NaN   
min           NaN            0.000000       NaN         NaN               NaN   
25%           NaN            3.000000       NaN         NaN               NaN   
50%           NaN            6.000000       NaN         NaN               NaN   
75%           NaN            9.000000       NaN         NaN               NaN   
max           NaN           12.000000       NaN         NaN               NaN   

                 ad_created  n_pictures   postal_code            last_seen  
count                 50000     50000.0  50000.000000                50000  
unique                   76         NaN           NaN                39481  
top     2016-04-03 00:00:00         NaN           NaN  2016-04-07 06:17:27  
freq                   1946         NaN           NaN                    8  
mean                    NaN         0.0  50813.627300                  NaN  
std                     NaN         0.0  25779.747957                  NaN  
min                     NaN         0.0   1067.000000                  NaN  
25%                     NaN         0.0  30451.000000                  NaN  
50%                     NaN         0.0  49577.000000                  NaN  
75%                     NaN         0.0  71540.000000                  NaN  
max                     NaN         0.0  99998.000000                  NaN

Above summary illustrates that 'seller' and 'offer_type' columns contain only one type of value for all records.
Anomalies:
'Price' contains some zero and NaN values. 'registration_year' also has a min value of 1000. There are Power_ps with zero values. ad_created contains value of 1946. 

In [0]:
autos['price'] = autos['price'].str.replace('$','').str.replace(',','').astype(float)

In [0]:
autos['odometer'] = autos['odometer'].str.replace('km','').str.replace(',','').astype(float)

In [0]:
autos.rename({'odometer':'odometer_km'}, axis=1, inplace=True)

We have converted 'price' and 'odometer' columns into a numeric datatype. This will facilitate our further analysis on outliers.

In [0]:
print(autos['price'].unique().shape)
print('\nhighest prices:\n\n',autos['price'].value_counts().sort_index(ascending=False).head(10))
print('\nlowest prices:\n\n',autos['price'].value_counts().sort_index().head(10))

autos['price'].describe()

(2357,)

highest prices:

 99999999.0    1
27322222.0    1
12345678.0    3
11111111.0    2
10000000.0    1
3890000.0     1
1300000.0     1
1234566.0     1
999999.0      2
999990.0      1
Name: price, dtype: int64

lowest prices:

 0.0     1421
1.0      156
2.0        3
3.0        1
5.0        2
8.0        1
9.0        1
10.0       7
11.0       2
12.0       3
Name: price, dtype: int64


count    5.000000e+04
mean     9.840044e+03
std      4.811044e+05
min      0.000000e+00
25%      1.100000e+03
50%      2.950000e+03
75%      7.200000e+03
max      1.000000e+08
Name: price, dtype: float64

Based on the outliers information we can discard the outliers since they will distort our analysis. Furthermore, some of the prices are unreasonable high or low. By keeping a certain threshold we make sure our analysis is reasonable.

In [0]:
autos = autos[autos['price'].between(500, 100000)]

In [0]:
print(autos['odometer_km'].unique().shape)
print('\nhighest km:\n\n',autos['odometer_km'].value_counts().sort_index(ascending=False).head(10))
print('\nlowest km:\n\n',autos['odometer_km'].value_counts().sort_index().head(10))

autos['odometer_km'].describe()

(13,)

highest km:

 150000.0    28695
125000.0     4836
100000.0     2029
90000.0      1676
80000.0      1384
70000.0      1189
60000.0      1129
50000.0       994
40000.0       807
30000.0       759
Name: odometer_km, dtype: int64

lowest km:

 5000.0      607
10000.0     234
20000.0     719
30000.0     759
40000.0     807
50000.0     994
60000.0    1129
70000.0    1189
80000.0    1384
90000.0    1676
Name: odometer_km, dtype: int64


count     45058.000000
mean     125364.752097
std       39542.959058
min        5000.000000
25%      100000.000000
50%      150000.000000
75%      150000.000000
max      150000.000000
Name: odometer_km, dtype: float64

Mileage data looks like within reasonable range: there are no huge or negative values. 

We have five date columns. In order to analyze them, we will need to check whether they also contain outliers.

In [0]:
autos[['date_crawled', 'ad_created', 'last_seen']][:5]

date_crawled           ad_created            last_seen
0  2016-03-26 17:47:46  2016-03-26 00:00:00  2016-04-06 06:45:54
1  2016-04-04 13:38:56  2016-04-04 00:00:00  2016-04-06 14:45:08
2  2016-03-26 18:57:24  2016-03-26 00:00:00  2016-04-06 20:15:37
3  2016-03-12 16:58:10  2016-03-12 00:00:00  2016-03-15 03:16:28
4  2016-04-01 14:38:50  2016-04-01 00:00:00  2016-04-01 14:38:50

In [0]:
autos['date_c rawled'].str[:10].value_counts(normalize=True, dropna=False).sort_index()

2016-03-05    0.025545
2016-03-06    0.014137
2016-03-07    0.036220
2016-03-08    0.033157
2016-03-09    0.032913
2016-03-10    0.032736
2016-03-11    0.033046
2016-03-12    0.037330
2016-03-13    0.015536
2016-03-14    0.036331
2016-03-15    0.034001
2016-03-16    0.029362
2016-03-17    0.031160
2016-03-18    0.012850
2016-03-19    0.034777
2016-03-20    0.038084
2016-03-21    0.037729
2016-03-22    0.032935
2016-03-23    0.032403
2016-03-24    0.028985
2016-03-25    0.031093
2016-03-26    0.032625
2016-03-27    0.031160
2016-03-28    0.034822
2016-03-29    0.033290
2016-03-30    0.033290
2016-03-31    0.031670
2016-04-01    0.033867
2016-04-02    0.035798
2016-04-03    0.038817
2016-04-04    0.036642
2016-04-05    0.013161
2016-04-06    0.003174
2016-04-07    0.001354
Name: date_crawled, dtype: float64

Most of the values for 'date_crawled' equally distributed for occured dates.

In [0]:
autos['ad_created'].str[:10].value_counts(normalize=True, dropna=False).sort_index()

2015-06-11    0.000022
2015-08-10    0.000022
2015-09-09    0.000022
2015-11-10    0.000022
2015-12-05    0.000022
2015-12-30    0.000022
2016-01-03    0.000022
2016-01-07    0.000022
2016-01-10    0.000044
2016-01-13    0.000022
2016-01-14    0.000022
2016-01-16    0.000022
2016-01-22    0.000022
2016-01-27    0.000067
2016-01-29    0.000022
2016-02-01    0.000022
2016-02-02    0.000044
2016-02-05    0.000044
2016-02-07    0.000022
2016-02-08    0.000022
2016-02-09    0.000022
2016-02-11    0.000022
2016-02-12    0.000044
2016-02-14    0.000044
2016-02-16    0.000022
2016-02-17    0.000022
2016-02-18    0.000044
2016-02-19    0.000067
2016-02-20    0.000044
2016-02-21    0.000044
                ...   
2016-03-09    0.033024
2016-03-10    0.032469
2016-03-11    0.033357
2016-03-12    0.037108
2016-03-13    0.016978
2016-03-14    0.034911
2016-03-15    0.033779
2016-03-16    0.029850
2016-03-17    0.030805
2016-03-18    0.013494
2016-03-19    0.033646
2016-03-20    0.038217
2016-03-21 

'ad_created' has many early dates representing only small proportion of the entire date column.

In [0]:
autos['last_seen'].str[:10].value_counts(normalize=True, dropna=False).sort_index()

2016-03-05    0.001087
2016-03-06    0.004172
2016-03-07    0.005216
2016-03-08    0.007013
2016-03-09    0.009454
2016-03-10    0.010298
2016-03-11    0.012051
2016-03-12    0.023925
2016-03-13    0.008877
2016-03-14    0.012295
2016-03-15    0.015691
2016-03-16    0.016157
2016-03-17    0.027698
2016-03-18    0.007390
2016-03-19    0.015425
2016-03-20    0.020418
2016-03-21    0.020662
2016-03-22    0.021261
2016-03-23    0.018399
2016-03-24    0.019508
2016-03-25    0.018598
2016-03-26    0.016423
2016-03-27    0.015447
2016-03-28    0.020529
2016-03-29    0.021372
2016-03-30    0.024169
2016-03-31    0.023459
2016-04-01    0.022882
2016-04-02    0.024879
2016-04-03    0.024946
2016-04-04    0.024324
2016-04-05    0.126481
2016-04-06    0.225354
2016-04-07    0.134138
Name: last_seen, dtype: float64

'last_seen' has dates that belong to 2016. Only one date seems to have more records: 2016-04-06: around 22.5% of the records are on this date. 

In [0]:
autos['registration_year'].value_counts().sort_index()

1000       1
1001       1
1910       2
1927       1
1929       1
1931       1
1934       2
1937       4
1938       1
1939       1
1941       2
1943       1
1948       1
1950       1
1951       1
1952       1
1953       1
1954       2
1955       2
1956       4
1957       2
1958       4
1959       6
1960      19
1961       6
1962       4
1963       8
1964      11
1965      17
1966      21
        ... 
2000    2714
2001    2519
2002    2419
2003    2657
2004    2683
2005    2864
2006    2666
2007    2269
2008    2204
2009    2078
2010    1582
2011    1616
2012    1304
2013     793
2014     650
2015     360
2016     944
2017    1296
2018     462
2019       1
2800       1
4100       1
4500       1
4800       1
5000       2
5911       1
6200       1
8888       1
9000       1
9999       3
Name: registration_year, Length: 93, dtype: int64

'registration_year' has several outliers: some values are around 1000 and above 2800.

Obviously, cars were not made in those years. We can safely remove those rows.

In [0]:
autos = autos[autos['registration_year'].between(1900,2016)]

In [0]:
autos['registration_year'].value_counts(normalize=True).sort_index()

1910    0.000046
1927    0.000023
1929    0.000023
1931    0.000023
1934    0.000046
1937    0.000092
1938    0.000023
1939    0.000023
1941    0.000046
1943    0.000023
1948    0.000023
1950    0.000023
1951    0.000023
1952    0.000023
1953    0.000023
1954    0.000046
1955    0.000046
1956    0.000092
1957    0.000046
1958    0.000092
1959    0.000139
1960    0.000439
1961    0.000139
1962    0.000092
1963    0.000185
1964    0.000254
1965    0.000393
1966    0.000485
1967    0.000601
1968    0.000578
          ...   
1987    0.001571
1988    0.002957
1989    0.003581
1990    0.006330
1991    0.006769
1992    0.007023
1993    0.007855
1994    0.011552
1995    0.019846
1996    0.024767
1997    0.034909
1998    0.046738
1999    0.059421
2000    0.062702
2001    0.058197
2002    0.055887
2003    0.061385
2004    0.061986
2005    0.066168
2006    0.061593
2007    0.052421
2008    0.050920
2009    0.048009
2010    0.036549
2011    0.037335
2012    0.030127
2013    0.018321
2014    0.0150

Next, let us analyze the 'brand column to see the top, average and lowest selling car brands.

In [0]:
autos['brand'].value_counts(sorted)

volkswagen        0.212088
bmw               0.115054
mercedes_benz     0.102278
opel              0.100430
audi              0.090565
ford              0.065197
renault           0.043942
peugeot           0.029387
fiat              0.023542
seat              0.017905
skoda             0.017166
smart             0.015133
mazda             0.015017
nissan            0.015017
citroen           0.014162
toyota            0.013423
hyundai           0.010165
sonstige_autos    0.009588
mini              0.009380
volvo             0.009380
honda             0.007809
mitsubishi        0.007740
kia               0.007347
alfa_romeo        0.006631
chevrolet         0.006030
suzuki            0.005914
porsche           0.005822
chrysler          0.003604
dacia             0.002842
jeep              0.002449
land_rover        0.002264
daihatsu          0.002172
subaru            0.001987
saab              0.001663
jaguar            0.001617
daewoo            0.001340
rover             0.001248
t

As we can see volkswagen, bmw and mecredes_benz are the most popular brands while trabant, lancia and lada are the least popular in German Ebay.

We will analyze the mean price of top 10 selling brands to see if they have substantial price differences.

In [0]:
top10 = list(autos['brand'].value_counts(sorted).index[:10])

In [0]:
mean_prices = {}
for brand in top10:
    one_brand = autos[autos['brand']==brand]
    mean_pr = one_brand['price'].mean()
    mean_prices[brand] = mean_pr

In [0]:
mean_prices

{'audi': 9571.457397959184,
 'bmw': 8447.069879518072,
 'fiat': 3256.152109911678,
 'ford': 4247.120481927711,
 'mercedes_benz': 8666.677208041563,
 'opel': 3394.0395675178283,
 'peugeot': 3360.9205974842766,
 'renault': 2819.059411146162,
 'seat': 4810.883870967742,
 'volkswagen': 5783.622984749455}

Based on the dataset analysis, the highest mean price for cars are of audi brand and then mercedes_benz brand with 9571 and 8666 accordingly. The lowest mean prices belong to renault and then fiat with 2819 and 3256 prices accordingly. 

Mean mileage for top 10 brands

In [0]:
mean_mileage = {}
for brand in top10:
    one_brand = autos[autos['brand']==brand]
    mean_km = one_brand['odometer_km'].mean()
    mean_mileage[brand] = mean_km
mean_mileage

{'audi': 128941.32653061225,
 'bmw': 132928.71485943775,
 'fiat': 114416.09421000982,
 'ford': 123520.55279943303,
 'mercedes_benz': 131083.12627061215,
 'opel': 128012.42236024844,
 'peugeot': 126073.11320754717,
 'renault': 126351.20925341746,
 'seat': 120058.06451612903,
 'volkswagen': 128234.74945533769}

In [0]:
mean_prices = pd.Series(mean_prices)
mean_mileage = pd.Series(mean_mileage)
price_km = pd.DataFrame(mean_prices, columns=["mean_price"])
price_km["mean_mileage"] = mean_mileage

print(price_km)

                mean_price   mean_mileage
audi           9571.457398  128941.326531
bmw            8447.069880  132928.714859
fiat           3256.152110  114416.094210
ford           4247.120482  123520.552799
mercedes_benz  8666.677208  131083.126271
opel           3394.039568  128012.422360
peugeot        3360.920597  126073.113208
renault        2819.059411  126351.209253
seat           4810.883871  120058.064516
volkswagen     5783.622985  128234.749455


Most of the cars from top 10 brands have over 110k km mileage and there are no substantial difference in their mileage on average. This means their prices are not influence by mileage. 